pandas/core/reshape/concat.py

```diff
@@ -585,6 +624,66 @@ def _maybe_check_integrity(self, concat_index: Index):
                     "{overlap!s}".format(overlap=overlap)
                 )
 
+    def _items_overlap_with_suffix(self, objs, suffixes):
+        """
+        Adding suffix for items if there is overlapping situation.
+
+        Be aware that `objs` can be either DataFrame-like or Index-like given
+        if `self._is_series` is True or False.
+        """
+        if self._is_series:
+
+            # when _is_series is True, objs are actually column Index
+            overlap_cols = list(objs)
+        else:
+            overlap_cols = chain.from_iterable([obj.columns for obj in objs])
+        to_rename = [col for col, cnt in Counter(overlap_cols).items() if cnt > 1]
+
+        if len(to_rename) == 0 or suffixes is None:
+            return objs
+
+        if not isinstance(suffixes, tuple):
+            raise ValueError(
+                "Invalid type {t} is assigned to suffixes, only"
+                " <class 'tuple'> is allowed.".format(t=type(suffixes))
```

Comment:
Prefer `str(tuple)` + f string - also mind adding a test case for this particular code path?

Reply:
i guess i already have a test case? or you mean something else?

Reply:
You have one for the ValueError below this using pytest.raises but not for this error?

Reply:
sorry, I didn't get your point. What do you mean by `have one for the ValueError below this`? I might misunderstand your words, could you please bear with me and clarify a bit? I had a test called `test_concat_suffixes_type `?

Reply:
Ahh yep thanks that test func covers it. I initially searched the test script for "Invalid type" as I assumed you would match on that - apologies look good

Reply:
ahh, i see, i used the other part for `match`. thanks for your comments!